crates/observe/src/heap_dump_handler.rs

```diff
@@ -0,0 +1,157 @@
+};
+
+/// Spawns a new async task that listens for connections to a UNIX socket
+/// at "/tmp/heap_dump_<process_name>.sock".
+/// When "dump" command is sent, it generates a heap profile using
+/// jemalloc_pprof and streams the binary protobuf data back through the socket.
+///
+/// Usage:
+/// ```bash
+/// # From your local machine (one-liner):
+/// kubectl exec <pod> -n <namespace> -- sh -c "echo dump | nc -U /tmp/heap_dump_<binary_name>.sock" > heap.pprof
+///
+/// # Analyze with pprof:
+/// go tool pprof -http=:8080 heap.pprof
+/// ```
+pub fn spawn_heap_dump_handler() {
+    // Check if jemalloc profiling is available before spawning the handler
+    // This prevents panics that would crash the entire process
+    let profiling_available =
+        std::panic::catch_unwind(|| jemalloc_pprof::PROF_CTL.as_ref().is_some()).unwrap_or(false);
+
+    if !profiling_available {
+        tracing::warn!(
+            "jemalloc profiling not available - heap dump handler not started. Ensure service is \
+             built with jemalloc-profiling feature and MALLOC_CONF is set."
+        );
+        return;
+    }
+
+    tokio::spawn(async move {
+        let name = binary_name().unwrap_or("unknown".to_string());
+        let socket_path = format!("/tmp/heap_dump_{name}.sock");
+
+        tracing::info!(socket = socket_path, "heap dump handler started");
+
+        let _ = tokio::fs::remove_file(&socket_path).await;
+        let listener = match UnixListener::bind(&socket_path) {
+            Ok(listener) => listener,
+            Err(err) => {
+                tracing::error!(
+                    ?err,
+                    socket = socket_path,
+                    "failed to bind heap dump socket"
+                );
+                return;
+            }
+        };
+        let handle = SocketHandle {
+            listener,
+            socket_path,
+        };
+
+        loop {
+            // Accept connection in main loop, then spawn task for each connection
+            // Sequential processing prevents multiple simultaneous expensive heap dumps
+            match handle.listener.accept().await {
+                Ok((socket, _addr)) => {
+                    let mut handle = tokio::spawn(async move {
+                        handle_connection_with_socket(socket).await;
+                    });
+
+                    // 5-minute timeout to prevent stuck dumps from blocking future requests
+                    match timeout(Duration::from_secs(300), &mut handle).await {
+                        Ok(Ok(())) => {
+                            // Task completed successfully
+                        }
+                        Ok(Err(err)) => {
+                            tracing::error!(?err, "panic in heap dump connection handler");
+                        }
+                        Err(_) => {
+                            handle.abort();
+                            tracing::error!("heap dump request timed out after 5 minutes");
```

Comment:
```suggestion
                            tracing::error!(?time, "heap dump request timed out");
```
Probably nice to store the timeout in a variable and reference it here so we don't run into the case that someone changes the timeout but not the log.
Also how long is this process expected to run? 5 minutes seems like a very long time.

Reply:
> Also how long is this process expected to run? 5 minutes seems like a very long time.

I am not sure. Reduced to 1 min, since the report should be ready in a few seconds.